server-timing/defaults.php

```diff
@@ -207,3 +207,84 @@ function( $acc, $query ) {
 	}
 }
 add_action( 'wp_loaded', 'perflab_register_default_server_timing_template_metrics' );
+
+/**
+ * Registers additional Server-Timing metrics as configured in the setting.
+ *
+ * These metrics should be registered as soon as possible. They can be added
+ * and modified in the "Tools > Server-Timing" screen.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_additional_server_timing_metrics_from_setting() {
+	$options = (array) get_option( PERFLAB_SERVER_TIMING_SETTING, array() );
+
+	if ( isset( $options['benchmarking_actions'] ) ) {
+		foreach ( $options['benchmarking_actions'] as $action ) {
+			$metric_slug = 'action-' . $action;
+
+			$measure_callback = function( $metric ) use ( $action ) {
+				$metric->measure_before();
+				add_action( $action, array( $metric, 'measure_after' ), PHP_INT_MAX, 0 );
```

Comment:
As add_action is just a wrapper around add_filter, can we just reuse the logic below.

Reply:
Good idea, I've made all that logic reusable now. It looks a bit more complicated now, but I've also provided additional inline documentation to explain, see https://github.com/WordPress/performance/pull/784/commits/694a6d1d4e3c968102ddce960ebbb252f468ed37.